src/main/java/com/microsoft/graph/authentication/BaseAuthenticationProvider.java

```diff
@@ -6,12 +6,35 @@
 import java.util.Locale;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 /**
  * Provides basic common methods for all authentication providers
  */
 public abstract class BaseAuthenticationProvider implements IAuthenticationProvider {
     private static final HashSet<String> validGraphHostNames = new HashSet<>(Arrays.asList("graph.microsoft.com", "graph.microsoft.us", "dod-graph.microsoft.us", "graph.microsoft.de", "microsoftgraph.chinacloudapi.cn", "canary.graph.microsoft.com"));
+    private HashSet<String> customHosts;
+
+    /**
+     * Allow the user to add custom hosts by passing in Array
+     * @param customHosts custom hosts passed in by user.
+     */
+    public void setCustomHosts(@Nonnull String[] customHosts) {
+        if(this.customHosts == null){
+            this.customHosts = new HashSet<String>();
+        }
+        for(String host: customHosts){
+            this.customHosts.add(host.toLowerCase(Locale.ROOT));
```

Comment:
nit: According to this [SO post](https://stackoverflow.com/questions/10336730/which-locale-should-i-specify-when-i-call-stringtolowercase?msclkid=42776460c25c11ec8cf90a5091d25b27), we should set this to Locale.ENGLISH in the very rare case we get to country specific endpoints (Turkey in the example) to avoid potential language specific user input error. With that stated, MG will likely define the host names in English so we should be safe even with Locale.ROOT.  

No action required as I don't see this scenario happening, just thought I'd share my learning.